EnoDatabase/EnoDatabase.cs

```diff
@@ -687,14 +687,82 @@ public async Task CalculateRoundTeamServiceStates(IServiceProvider serviceProvid
+            var currentRoundWorstResults = new Dictionary<(long ServiceId, long TeamId), CheckerTask?>();
+            var sw = new Stopwatch();
+            sw.Start();
+            foreach (var t in teams)
+                foreach (var s in services)
+            {
+                    currentRoundWorstResults[(s.Id, t.Id)] =
+                        await _context.CheckerTasks
+                            .TagWith("CalculateRoundTeamServiceStates:currentRoundTasks")
+                            .Where(ct => ct.CurrentRoundId == roundId)
+                            .Where(ct => ct.RelatedRoundId == roundId)
+                            .Where(ct => ct.TeamId == t.Id)
+                            .Where(ct => ct.ServiceId == s.Id)
+                            .Where(ct => ct.CheckerResult == CheckerResult.INTERNAL_ERROR)
+                            .OrderBy(ct => ct.StartTime)
+                            .Select(ct => new CheckerTask()
+                            {
+                                CheckerResult = ct.CheckerResult,
+                                ErrorMessage = ct.ErrorMessage
+                            })
+                            .FirstOrDefaultAsync() ??
+                                                await _context.CheckerTasks
+                            .TagWith("CalculateRoundTeamServiceStates:currentRoundTasks")
+                            .Where(ct => ct.CurrentRoundId == roundId)
+                            .Where(ct => ct.RelatedRoundId == roundId)
+                            .Where(ct => ct.TeamId == t.Id)
+                            .Where(ct => ct.ServiceId == s.Id)
+                            .Where(ct => ct.CheckerResult == CheckerResult.OFFLINE)
+                            .OrderBy(ct => ct.StartTime)
+                            .Select(ct => new CheckerTask()
+                            {
+                                CheckerResult = ct.CheckerResult,
+                                ErrorMessage = ct.ErrorMessage
+                            })
+                            .FirstOrDefaultAsync() ??
+                                                    await _context.CheckerTasks
+                            .TagWith("CalculateRoundTeamServiceStates:currentRoundTasks")
+                            .Where(ct => ct.CurrentRoundId == roundId)
+                            .Where(ct => ct.RelatedRoundId == roundId)
+                            .Where(ct => ct.TeamId == t.Id)
+                            .Where(ct => ct.ServiceId == s.Id)
+                            .Where(ct => ct.CheckerResult == CheckerResult.MUMBLE)
+                            .OrderBy(ct => ct.StartTime)
+                            .Select(ct => new CheckerTask()
+                            {
+                                CheckerResult = ct.CheckerResult,
+                                ErrorMessage = ct.ErrorMessage
+                            })
+                            .FirstOrDefaultAsync() ?? null;

```

Comment:
How does this impact performance? Last year we changed away from that, but the testing hw was really old and had no SSD

Reply:
On the Testsetup at hetzner it takes quite some time (~8s), so some optimisation is needed. But I suspect the index is still bit correct, that might be a problem. I tried to fix the index, but I am not sure it is used correctly

Reply:
I suspect that you have to transport way more data per db roundtrip, sequentially doing n*m requests accumulates a lot of per-transaction overhead.